### PR TITLE
feat: add multi-day event support (Phase B)

### PR DESCRIPTION
**Pull request by joe-bor**
## Summary

- Add `endDate` field to calendar event types, service mapping, Zod validation, and form UI
- Add `isEventOnDate()` utility for multi-day range checking used by all 4 calendar views
- Multi-day events render as repeated all-day badges on each day in the range (daily, weekly, monthly, schedule views)
- End date picker appears only when "All day" toggle is on, with `fromDate` constraint preventing selection before start date
- Event detail modal shows date range (e.g., "March 8 – March 12, 2026") for multi-day events
- Optimistic updates include `endDate` mapping; `null` sent to API to explicitly clear endDate

## Test plan

- [x] `npm run build` passes (type-check + build)
- [x] `npm run test -- --run` passes (467 tests, including new multi-day tests)
- [x] Manual: create multi-day all-day event, verify it appears on each day in all 4 views
- [x] Manual: edit multi-day event, toggle isAllDay off, verify endDate clears
- [x] Manual: verify end date picker prevents selecting dates before start date

### CODE REVIEW

**Comment by joe-bor** [comment]
## PR Review: feat: add multi-day event support (Phase B)

### Summary
Adds `endDate` support for all-day calendar events, enabling multi-day events that render across all four calendar views. Includes type changes, Zod validation with cross-field refinements, form UI with constrained end date picker, and solid test coverage.

### Verdict: Approve (with minor suggestions)

This is a clean, well-structured feature. The implementation follows established patterns, has good test coverage, and the commit history tells a clear story.

### Strengths
- **Centralized `isEventOnDate()` utility** — single function in `time-utils.ts` used by all 4 views, eliminates duplicated date-matching logic
- **Good Zod validation** — cross-field refinements for `endDate >= date` and `endDate` only valid for all-day events
- **`fromDate` constraint** on the end date picker prevents invalid selection via UI
- **Thorough test coverage** — unit tests for `isEventOnDate`, validation schema tests for all endDate edge cases, and form integration tests including toggle behavior
- **Consistent patterns** — follows existing form, type, and service layer conventions
- **Clean commit history** — atomic commits progressing logically through the stack

### Issues (Must Fix)
None — no blocking issues found.

### Suggestions (Non-blocking)

**1. `isEventOnDate` could normalize dates for safety** (`src/lib/time-utils.ts:123`)
The range check `targetDate >= event.date && targetDate <= event.endDate` compares timestamps. This works because `parseLocalDate` produces start-of-day dates, but the function itself doesn't enforce this invariant. If any caller passes a Date with a non-midnight time, the comparison could produce incorrect results. Consider normalizing to date-only comparison:
```ts
const targetStart = new Date(targetDate.getFullYear(), targetDate.getMonth(), targetDate.getDate());
const eventStart = new Date(event.date.getFullYear(), event.date.getMonth(), event.date.getDate());
const eventEnd = new Date(event.endDate.getFullYear(), event.endDate.getMonth(), event.endDate.getDate());
return targetStart >= eventStart && targetStart <= eventEnd;
```
Low priority since all current callers are safe, but it makes the function defensive.

**2. Stale endDate when start date changes** (`src/components/calendar/components/event-form.tsx:173`)
If a user sets endDate to March 10, then changes the start date to March 12, the endDate "2026-03-10" remains in form state. The `fromDate` picker constraint updates visually, and Zod catches it on submit, but the user sees no immediate feedback. Consider clearing endDate when the start date moves past it, or triggering validation on date change.

**3. Cross-year date range display** (`src/components/calendar/components/event-detail-modal.tsx:64`)
`format(event.date, "MMMM d") – format(event.endDate, "MMMM d, yyyy")` — if an event spans Dec 31 → Jan 2, the start date won't show its year (displays "December 31 – January 2, 2027"). Consider showing year on both dates when they differ, or always showing year on both.

**4. Monthly view O(events × days) complexity** (`src/components/calendar/views/monthly-calendar.tsx:96`)
The old single-pass approach was O(events). The new nested loop is O(events × days) (~42 days for monthly). For household-scale event counts this is negligible, but noting the tradeoff. An alternative would be to expand multi-day events into date keys once, then do the single-pass lookup — but this is premature optimization for the current use case.

### Technical Debt
No new technical debt introduced. The implementation is focused and avoids over-engineering.

File: src/lib/time-utils.ts [hunk 1]
```diff
@@ -109,6 +109,20 @@ export function parseLocalDate(dateStr: string): Date {
...
+ * Handles both single-day events (exact date match) and multi-day events (date range).
+ */
+export function isEventOnDate(
+  event: { date: Date; endDate?: Date },
+  targetDate: Date,
+): boolean {
+  const target = targetDate.toDateString();
+  if (event.date.toDateString() === target) return true;
+  if (!event.endDate) return false;
+  return targetDate >= event.date && targetDate <= event.endDate;
```
Comment (joe-bor) [inline src/lib/time-utils.ts]: Minor robustness suggestion: This range comparison works correctly because all callers pass start-of-day dates (via `parseLocalDate`), but the function itself doesn't enforce this invariant. If a future caller passes a Date with non-midnight time, the `>=`/`<=` comparison could give incorrect results. Consider normalizing to date-only comparison inside the function for defensiveness. Low priority since all current callers are safe.

File: src/components/calendar/components/event-form.tsx [hunk 2]
```diff
@@ -161,6 +164,26 @@ function EventForm({
...
+      {/* End Date (multi-day all-day events) */}
+      {isAllDayValue && (
+        <div className="space-y-2">
+          <Label>End Date</Label>
+          <DatePicker
+            value={endDateAsDate}
+            onChange={(date) => {
+              setValue(
+                "endDate",
+                date ? format(date, "yyyy-MM-dd") : undefined,
```
Comment (joe-bor) [inline src/components/calendar/components/event-form.tsx:180]: Edge case: if the user changes the start date to be *after* the currently selected end date, the endDate value remains stale in form state. The `fromDate` constraint updates on the picker, and Zod catches it on submit, but there's no immediate feedback. Consider clearing `endDate` (or triggering validation) when the start date changes past it.

File: src/components/calendar/components/event-detail-modal.tsx [hunk 3]
```diff
@@ -60,8 +60,10 @@ function EventDetailModal({
     onDelete();
   };
 
-  // Format date for display (e.g., "Monday, December 23, 2025")
-  const formattedDate = format(event.date, "EEEE, MMMM d, yyyy");
+  // Format date for display — multi-day shows range, single-day shows full date
+  const formattedDate = event.endDate
```
Comment (joe-bor) [inline src/components/calendar/components/event-detail-modal.tsx:64]: Minor: if an event spans years (e.g., Dec 31 → Jan 2), only the end date shows the year. Consider showing year on both dates when `event.date` and `event.endDate` are in different years.
